### PR TITLE
Feature/point spread function

### PR DESCRIPTION
**Pull request by Kmudsam01**
* **Tickets addressed:** 509
* **Review:** By commit 
* **Merge strategy:** Merge (no squash) 

## Description
This PR implements a point spread function using a gaussian blur. It can read/write files relative to the project directory, but is not yet connected to Basilisk messaging system. Right now, once the simulation is running and the user presses "1", it reads test_image0.png in cielim/Test_Images, applies the points spread function, and writes the resulting image, PSF.jpg, to cielem/Result_Images.

## Verification
The changes were validated by using certain test images found in cielim/Test_Images, which clearly display the blurring made by the gaussian blur.

## Documentation
No documentation is affected.

## Future work
Future work includes connecting the parameters specified by a basilisk message to ApplyPSF_Gaussian, the point spread function implementation.

### CODE REVIEW

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp [hunk 1]
```diff
@@ -1,28 +1,62 @@
...
 void URenderingFunctionsLibrary::ApplyReadNoise()
 {
-	
+	//Do Not Call
+	throw std::logic_error("Function not yet implemented");
 }
+
+
+
+
```
Comment (thibaudteil) [inline Source/cielim/Private/RenderingFunctionsLibrary.cpp]: You can delete these extra lines

File: Source/cielim/Private/RenderingFunctionsLibrary.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	static void ApplyReadNoise();
+
+	static void ReadImage(){};
+
+	
+
+	
+};
+
+
```
Comment (thibaudteil) [inline Source/cielim/Private/RenderingFunctionsLibrary.h]: remove extra lines

**Review by thibaudteil — APPROVED** [review]
Just some minor cleanup!

File: Source/cielim/Private/RenderingFunctionsLibrary.h (hunk 2 above)
Comment (Kmudsam01) [reply Source/cielim/Private/RenderingFunctionsLibrary.h]: Aye Captain!

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp [hunk 3]
```diff
@@ -1,28 +1,58 @@
...
 {
+	//NOTE: both dimensions of KernelSize must be odd
+	//Read Image
+
+	FString Filepath_Absolute = IFileManager::Get().ConvertToAbsolutePathForExternalAppForRead(*Filepath);
+	
+	std::string Filepath_Absolute_String = TCHAR_TO_UTF8(*Filepath_Absolute);
+	cv::Mat Image = cv::imread(Filepath_Absolute_String);
+
+	//Init ResultImage
```
Comment (patkenneally) [inline Source/cielim/Private/RenderingFunctionsLibrary.cpp]: The attention to adding comments is great. However, let's be careful to only add comments when it aids understanding.

File: Result_Images/PSF.jpg
Comment (patkenneally) [inline Result_Images/PSF.jpg:1]: I'm not seeing the reason for adding the resultant image to the repo. can you help me understand why this is committed?

**Review by patkenneally — APPROVED** [review]

File: Result_Images/PSF.jpg
Comment (Kmudsam01) [reply Result_Images/PSF.jpg:1]: Ah, it really shouldn't have, I'll remove it from the repo

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp (hunk 3 above)
Comment (Kmudsam01) [reply Source/cielim/Private/RenderingFunctionsLibrary.cpp]: Yes! I'll remove some of the more redundant comments

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp [hunk 4]
```diff
@@ -1,28 +1,58 @@
...
+	throw std::logic_error("Function not yet implemented");
 }
 
 void URenderingFunctionsLibrary::ApplyReadNoise()
 {
-	
-}
+	//Do Not Call
+	throw std::logic_error("Function not yet implemented");
+}
```
Comment (patkenneally) [inline Source/cielim/Private/RenderingFunctionsLibrary.cpp]: :small_red_triangle_down: Files should have a single trailing new line at the end. Github indicates this by showing the red circle and line to the left.

https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: Source/cielim/Private/RenderingFunctionsLibrary.h [hunk 5]
```diff
@@ -20,23 +17,11 @@ class  CIELIM_API URenderingFunctionsLibrary :
...
-	static void ReadImage(){};
-
-	
-
-	
-};
-
-
-
+};
```
Comment (patkenneally) [inline Source/cielim/Private/RenderingFunctionsLibrary.h]: :small_red_triangle_down: New line here also.

File: Source/cielim/Private/RenderingFunctionsLibrary.h [hunk 6]
```diff
@@ -4,10 +4,7 @@
 #include "Math/Vector.h"
 #include "Engine/World.h"
 #include "Kismet/BlueprintFunctionLibrary.h"
-
-#include <OpenCV/PreOpenCVHeaders.h>
-#include "OpenCV/opencv-4.5.5/modules/imgcodecs/include/opencv2/imgcodecs.hpp"
-#include <OpenCV/PostOpenCVHeaders.h>
+#include <string>
```
Comment (patkenneally) [inline Source/cielim/Private/RenderingFunctionsLibrary.h]: :small_red_triangle_down: This include is unused and should be removed.

**Review by patkenneally — APPROVED** [review]
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: Source/cielim/Private/RenderingFunctionsLibrary.h (hunk 6 above)
Comment (Kmudsam01) [reply Source/cielim/Private/RenderingFunctionsLibrary.h]: #include \<string\> removed

File: Source/cielim/Private/RenderingFunctionsLibrary.h (hunk 5 above)
Comment (Kmudsam01) [reply Source/cielim/Private/RenderingFunctionsLibrary.h]: Line also added!

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp (hunk 4 above)
Comment (Kmudsam01) [reply Source/cielim/Private/RenderingFunctionsLibrary.cpp]: Line added!

**Pull request merged**